test/system/065-cp.bats

```diff
@@ -7,6 +7,312 @@
 
 load helpers
 
+@test "podman cp file from host to container" {
+    skip_if_remote "podman-remote does not yet handle cp"
+
+    srcdir=$PODMAN_TMPDIR/cp-test-file-host-to-ctr
+    mkdir -p $srcdir
+    # Create multiple files which prevents some (expensive) clean ups via
+    # podman-exec.
+    echo "This file is on the host" > $srcdir/hostfile
+    echo "This file is on the host (1)" > $srcdir/hostfile1
+    echo "This file is on the host (2)" > $srcdir/hostfile2
+
+    run_podman run -d --name cpcontainer --workdir=/srv $IMAGE sleep infinity
+
+    # Copy to root.
+    run_podman cp $srcdir/hostfile cpcontainer:/
+    run_podman exec cpcontainer cat /hostfile
+    is "$output" "This file is on the host"
+
+    run_podman cp $srcdir/hostfile cpcontainer:/anotherbase.txt
+    run_podman exec cpcontainer cat /anotherbase.txt
+    is "$output" "This file is on the host"
+
+    # Copy to a non-root directory (varied input syntax).
+    run_podman cp $srcdir/hostfile cpcontainer:/tmp
+    run_podman exec cpcontainer cat /tmp/hostfile
+    is "$output" "This file is on the host"
+
+    run_podman cp $srcdir/hostfile1 cpcontainer:/tmp/
+    run_podman exec cpcontainer cat /tmp/hostfile1
+    is "$output" "This file is on the host (1)"
+    run_podman exec cpcontainer rm /tmp/hostfile
+
+    run_podman cp $srcdir/hostfile2 cpcontainer:/tmp/.
+    run_podman exec cpcontainer cat /tmp/hostfile2
+    is "$output" "This file is on the host (2)"
+
+    run_podman cp $srcdir/hostfile cpcontainer:/tmp/anotherbase.txt
+    run_podman exec cpcontainer cat /tmp/anotherbase.txt
+    is "$output" "This file is on the host"
+
+    # Copy to a relative path -> must resolve to the container's work dir
+    # (i.e., /srv).
+    run_podman exec cpcontainer mkdir /srv/subdir
+    run_podman cp $srcdir/hostfile cpcontainer:subdir
+    run_podman exec cpcontainer cat /srv/subdir/hostfile
+    is "$output" "This file is on the host"
+
+    run_podman cp $srcdir/hostfile cpcontainer:anotherbase.txt
+    run_podman exec cpcontainer cat /srv/anotherbase.txt
+    is "$output" "This file is on the host"
+
+    # Host path does not exist.
+    run_podman 125 cp $srcdir/IdoNotExist cpcontainer:/tmp
+    is "$output" 'Error: ".*/IdoNotExist" could not be found on the host'
+
+    # Container path does not exist.  Notice that the error message shows how
+    # the specified container is resolved.
+    run_podman 125 cp $srcdir/hostfile cpcontainer:/IdoNotExist/
+    is "$output" 'Error: "/IdoNotExist/" could not be found on container.*(resolved to .*/IdoNotExist.*'
+
+    run_podman rm -f cpcontainer
+}
```

Comment:
Any time I see a huge list of commands with only very teeny differences between them, I think: time for a table. Friendly suggestion:
```diff
diff --git a/test/system/065-cp.bats b/test/system/065-cp.bats
index 789267fcf..8e993ecf4 100644
--- a/test/system/065-cp.bats
+++ b/test/system/065-cp.bats
@@ -14,58 +14,54 @@ load helpers
     mkdir -p $srcdir
     # Create multiple files which prevents some (expensive) clean ups via
     # podman-exec.
-    echo "This file is on the host" > $srcdir/hostfile
-    echo "This file is on the host (1)" > $srcdir/hostfile1
-    echo "This file is on the host (2)" > $srcdir/hostfile2
+    local -a randomcontent=(
+        random-0-$(random_string 10)
+        random-1-$(random_string 15)
+        random-2-$(random_string 20)
+    )
+    echo "${randomcontent[0]}" > $srcdir/hostfile0
+    echo "${randomcontent[1]}" > $srcdir/hostfile1
+    echo "${randomcontent[2]}" > $srcdir/hostfile2

     run_podman run -d --name cpcontainer --workdir=/srv $IMAGE sleep infinity
-
-    # Copy to root.
-    run_podman cp $srcdir/hostfile cpcontainer:/
-    run_podman exec cpcontainer cat /hostfile
-    is "$output" "This file is on the host"
-
-    run_podman cp $srcdir/hostfile cpcontainer:/anotherbase.txt
-    run_podman exec cpcontainer cat /anotherbase.txt
-    is "$output" "This file is on the host"
-
-    # Copy to a non-root directory (varied input syntax).
-    run_podman cp $srcdir/hostfile cpcontainer:/tmp
-    run_podman exec cpcontainer cat /tmp/hostfile
-    is "$output" "This file is on the host"
-
-    run_podman cp $srcdir/hostfile1 cpcontainer:/tmp/
-    run_podman exec cpcontainer cat /tmp/hostfile1
-    is "$output" "This file is on the host (1)"
-    run_podman exec cpcontainer rm /tmp/hostfile
-
-    run_podman cp $srcdir/hostfile2 cpcontainer:/tmp/.
-    run_podman exec cpcontainer cat /tmp/hostfile2
-    is "$output" "This file is on the host (2)"
-
-    run_podman cp $srcdir/hostfile cpcontainer:/tmp/anotherbase.txt
-    run_podman exec cpcontainer cat /tmp/anotherbase.txt
-    is "$output" "This file is on the host"
-
-    # Copy to a relative path -> must resolve to the container's work dir
-    # (i.e., /srv).
     run_podman exec cpcontainer mkdir /srv/subdir
-    run_podman cp $srcdir/hostfile cpcontainer:subdir
-    run_podman exec cpcontainer cat /srv/subdir/hostfile
-    is "$output" "This file is on the host"

-    run_podman cp $srcdir/hostfile cpcontainer:anotherbase.txt
-    run_podman exec cpcontainer cat /srv/anotherbase.txt
-    is "$output" "This file is on the host"
+    # format is: <id> | <destination arg to cp> | <full dest path> | <test name>
+    # where:
+    #    id        is 0-2, one of the random strings/files
+    #    dest arg  is the right-hand argument to 'podman cp' (may be implicit)
+    #    dest path is the full explicit path we expect to see
+    #    test name is a short description of what we're testing here
+    tests="
+0 | /                    | /hostfile0            | copy to root
+0 | /anotherbase.txt     | /anotherbase.txt      | copy to root, new name
+0 | /tmp                 | /tmp/hostfile0        | copy to /tmp
+1 | /tmp                 | /tmp/hostfile1        | copy to /tmp
+2 | /tmp                 | /tmp/hostfile2        | copy to /tmp
+0 | /tmp/hostfile2       | /tmp/hostfile2        | overwrite previous copy
+0 | /tmp/anotherbase.txt | /tmp/anotherbase.txt  | copy to /tmp, new name
+0 | anotherbase.txt      | /srv/anotherbase.txt  | copy to workdir, new name
+0 | subdir               | /srv/subdir/hostfile0 | copy to workdir/subdir
+"
+
+    # Copy one of the files into container, exec+cat, confirm the file
+    # is there and matches what we expect
+    while read id dest dest_fullname description; do
+        run_podman cp $srcdir/hostfile$id cpcontainer:$dest
+        run_podman exec cpcontainer cat $dest_fullname
+        is "$output" "${randomcontent[$id]}" "$description (cp -> ctr:$dest)"
+    done < <(parse_table "$tests")

     # Host path does not exist.
     run_podman 125 cp $srcdir/IdoNotExist cpcontainer:/tmp
-    is "$output" 'Error: ".*/IdoNotExist" could not be found on the host'
+    is "$output" 'Error: ".*/IdoNotExist" could not be found on the host' \
+       "copy nonexistent host path"

     # Container path does not exist.  Notice that the error message shows how
     # the specified container is resolved.
-    run_podman 125 cp $srcdir/hostfile cpcontainer:/IdoNotExist/
-    is "$output" 'Error: "/IdoNotExist/" could not be found on container.*(resolved to .*/IdoNotExist.*'
+    run_podman 125 cp $srcdir/hostfile0 cpcontainer:/IdoNotExist/
+    is "$output" 'Error: "/IdoNotExist/" could not be found on container.*(resolved to .*/IdoNotExist.*' \
+       "copy into nonexistent path in container"

     run_podman rm -f cpcontainer
 }
```
[cp-test.txt](https://github.com/containers/podman/files/5637173/cp-test.txt)

Reply:
Thanks add!

I can apply this diff if you want me to.  My bash skills are limited, so I tend to play it safe :^)

Reply:
If you agree that it's more maintainable, then yes, please apply it. (You might notice that I added one extra test that was not in the original. IT was a one-line addition instead of multiple error-prone ones).

Reply:
I like @edsantiago idea, apply the diff.

Reply:
It works like a charm and is much more readable. Thanks, Ed!